### PR TITLE
[MISC] Make literals conditionally constexpr

### PR DESCRIPTION
**Pull request by eseiler**
Resolves https://github.com/seqan/product_backlog/issues/399

https://godbolt.org/z/TefTaea88

- [x] Changelog

### CODE REVIEW

**Review by remyschwab — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
   * Added `seqan3::views::char_strictly_to`. Behaves like `seqan3::views::char_to`, but throws on invalid
     input ([\#2898](https://github.com/seqan/seqan3/pull/2898)).
   * Improved performance of vector assignment for alphabets ([\#3038](https://github.com/seqan/seqan3/pull/3038)).
+  * Char literals returning std::vector are now constexpr if supported by the compiler
```
Comment (eseiler) [inline CHANGELOG.md:36]: I'll move this to the proper release when rebasing

Comment (smehringer) [reply CHANGELOG.md:36]: rebase please :)

Comment (smehringer) [reply CHANGELOG.md:36]: rebase please

**Review by smehringer — CHANGES_REQUESTED** [review]
only the rebase missing. I'' check the change log then

**Pull request merged**